consensus/service/src/consensus_service.rs

```diff
@@ -636,70 +646,165 @@ impl<
+                                Some("ByzantineLedger has been dropped".to_string()),
+                            ))
+                        }
+                    }?;
+
+                let byzantine_ledger_instance = match byzantine_ledger_arc_once_cell.get() {
+                    Some(instance) => Ok(instance),
+                    None => {
+                        log::error!(logger, "ByzantineLedger has not been initialized.");
+                        Err(RpcStatus::new(
+                            RpcStatusCode::UNAVAILABLE,
+                            Some("ByzantineLedger has not been initialized".to_string()),
+                        ))
+                    }
+                }?;
+
+                let is_behind: bool = byzantine_ledger_instance.is_behind();
+                let highest_peer_block_index: u64 = byzantine_ledger_instance.highest_peer_block();
+                (is_behind, highest_peer_block_index)
             };
-            Ok(json!({
+
+            let network_config = config.network();
+
+            let json: String = config_and_status_as_json(
+                &config,
+                &network_config,
+                &highest_block,
+                highest_block_signature.as_ref(),
+                is_behind,
+                highest_peer_block_index,
+            );
+
+            Ok(json)
+        })
+    }
+}
+
+impl<
+        E: ConsensusEnclave + Clone + Send + Sync + 'static,
+        R: RaClient + Send + Sync + 'static,
+        TXM: TxManager + Clone + Send + Sync + 'static,
+    > Drop for ConsensusService<E, R, TXM>
+{
+    fn drop(&mut self) {
+        let _ = self.stop();
+    }
+}
+
+/// Format the service's configuration and ledger status as JSON.
+///
+/// # Arguments
+/// * `config` - ConsensusService configurations.
+/// * `network_config` - Network configuration.
+/// * `highest_block` - Highest block in the local ledger.
+/// * `highest_block_signature` - Signature, if any, for the highest block in the local ledger.
+/// * `is_behind` - True if the local ledger is behind the peers' ledgers.
+/// * `highest_peer_block_index` -  Highest block index agreed upon by peers.
+fn config_and_status_as_json(
+    config: &Config,
+    network_config: &NetworkConfig,
+    highest_block: &Block,
+    highest_block_signature: Option<&BlockSignature>,
+    is_behind: bool,
+    highest_peer_block_index: u64,
+) -> String {
+    // The approximate time in which the block was signed, represented at seconds
+    //  of UTC time since Unix epoch 1970-01-01T00:00:00Z.
+    let highest_block_signed_at: Option<u64> =
+        highest_block_signature.map(|signature| signature.signed_at());
+
+    let sync_status = if is_behind { "catchup" } else { "synced" };
+
+    // Number of blocks that the local node is behind, or zero if the local node is not behind.
+    // (It's funky that this is both an Option and a saturating value. It would probably be simpler to
```

Comment:
Is it OK to change the JSON structure? So far, I've kept the JSON fields as-is, but I think there's room for improvement.

Reply:
Yeah absolutely